chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/developer/DeveloperInSystemIsSavedValidator.java

```diff
@@ -0,0 +1,114 @@
+package gov.healthit.chpl.validation.developer;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+
+import gov.healthit.chpl.dto.DeveloperACBMapDTO;
+import gov.healthit.chpl.dto.DeveloperDTO;
+import gov.healthit.chpl.util.ErrorMessageUtil;
+
+/**
+ * Validate fields used when creating a new developer.
+ */
+@Component("developerInSystemIsSavedValidator")
```

Comment:
This is the part I think I'd like to talk about. There's already a Validator for Developer creation, and one for Developer Update. I know the Update Validator can't require all data, due to legacy data, but the Create one can. The Create one is also the one that's used when a new Developer is created, which only happens on Listing upload, so maybe that Create one should be expanded / tweaked to basically be something like "DeveloperDuringUploadValidator". Maybe the "DeveloperCreateValidator" should actually go away, as we can't POST a new Developer at all anyway. There's no way for a Developer to exist in the CHPL without having a Product, that Product having a Version, and that Version having a Listing, so creating just a Developer on its own is impossible

Reply:
Thanks for the review. A discussion on this would be great. Some good points. My thoughts since it's on my mind:
I designed DeveloperInSystemIsSavedValidator to only care about if the data in the system exists (with the exception of attestation), not that it is valid so much. DeveloperCreationValidator handles whether it's valid or not (well, a subset of it anyway). That's how I am able to use it that way, because I know if the data is there, it must be valid. So they have separate concerns. The classes are simple in that they just have one validate method so extending and overriding doesn't really give any code-reuse. So I separated. However, as for having one merged version of the two classes (e.g. DeveloperDuringUploadValidator) handle it all, that could certainly work, and would require less code and may be less confusing/less error-prone. So that seems like a good idea. Although, it would cause the missing info use case to check more than is needed. Again, we would know that stored data is valid as it's been through the process before. The performance though is negligible. and not an argument. I see your point. 

The main reasons that I went with my own class were: 
1: The requirements were different than what I had to support. For example, DeveloperCreationValidator does not require a website. It doesn't require a transparency attestation or validate it in any way. So this is by far the number 1 reason because I didn't think I was allowed to enforce things that were not already enforced. But it appears I am in this case? If so, merging into one class is likely the best solution.
2: I wanted to make it obvious through my own message properties that the error was related to bad data that is already in the system vs trying to store brand new bad data. Of course, for point 2, I could find another way to implement such a distinction but would need to avoid the possible pitfall of it being messy since confined to one class. Or, I could just not provide the distinction at all as the importance of separate properties (excepting the literal extra ones) is probably quite low.